packages/language-server/src/edit/handlers/moveLinePosHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { IncrementalUpdate, MoveLposEdit } from "@hylimo/diagram-protocol";
+import { EditHandler } from "./editHandler.js";
+import { DefaultEditTypes, LinePoint } from "@hylimo/diagram-common";
+import { roundToPrecision } from "../../util/roundToPrecision.js";
+
+/**
+ * Handler for lpos pos edits
+ */
+export const moveLinePosHandler: EditHandler<MoveLposEdit> = {
+    type: DefaultEditTypes.MOVE_LPOS_POS,
+
+    predictActionDiff(layoutedDiagram, lastApplied, newest, elements) {
+        let pos: number;
+        let segment: number | undefined;
+        const newestPos = newest.pos;
+        if (Array.isArray(newestPos)) {
+            segment = newestPos[0];
+            pos = newestPos[1];
+        } else {
+            pos = newestPos;
+            segment = undefined;
+        }
+        const updates: IncrementalUpdate[] = [];
+        for (const element of elements) {
+            if (LinePoint.isLinePoint(element)) {
+                element.pos = pos;
+                element.segment = segment;
+                updates.push({
+                    target: element.id,
+                    changes: {
+                        pos: element.pos,
```

Comment:
this is not about a segment, this is just about a single point definition
it can be used for example for a start/end point of a canvas connection segment, but also as the position for a canvas element, or the target of a relative point